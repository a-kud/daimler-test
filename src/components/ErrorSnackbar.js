import React, { useState } from 'react'
import CloseIcon from '@material-ui/icons/Close'
import Snackbar from '@material-ui/core/Snackbar'
import IconButton from '@material-ui/core/IconButton'
import { makeStyles } from '@material-ui/core'

const useStyles = makeStyles({
  root: {
    backgroundColor: 'red'
  }
})

const ErrorSnackbar = ({ message }) => {
  const classes = useStyles()
  const [open, setOpen] = useState(Boolean(message))

  const handleClose = (event, reason) => {
    if (reason === 'clickaway') {
      return
    }

    setOpen(false)
  }

  return (
    <>
      <Snackbar
        anchorOrigin={{
          vertical: 'bottom',
          horizontal: 'center'
        }}
        open={open}
        autoHideDuration={6000}
        onClose={handleClose}
        ContentProps={{
          'aria-describedby': 'message-id',
          classes: {
            root: classes.root
          }
        }}
        message={<span id='message-id'>{message}</span>}
        action={[
          <IconButton
            key='close'
            aria-label='close'
            color='inherit'
            onClick={handleClose}
          >
            <CloseIcon />
          </IconButton>
        ]}
      />
    </>
  )
}

export default ErrorSnackbar
