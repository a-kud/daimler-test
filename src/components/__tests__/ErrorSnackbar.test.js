import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import ErrorSnackbar from '../ErrorSnackbar'

describe('ErrorSnackbar Component', () => {
  const wrapper = shallow(<ErrorSnackbar />)

  it('should render correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
