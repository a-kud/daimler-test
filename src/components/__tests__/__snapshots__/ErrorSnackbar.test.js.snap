// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorSnackbar Component should render correctly 1`] = `
<Fragment>
  <WithStyles(ForwardRef(Snackbar))
    ContentProps={
      Object {
        "aria-describedby": "message-id",
        "classes": Object {
          "root": "makeStyles-root-1",
        },
      }
    }
    action={
      Array [
        <ForwardRef(WithStyles)
          aria-label="close"
          color="inherit"
          onClick={[Function]}
        >
          <UNDEFINED />
        </ForwardRef(WithStyles)>,
      ]
    }
    anchorOrigin={
      Object {
        "horizontal": "center",
        "vertical": "bottom",
      }
    }
    autoHideDuration={6000}
    message={
      <span
        id="message-id"
      />
    }
    onClose={[Function]}
    open={false}
  />
</Fragment>
`;
